RA-1731 : Update Reference Metadata Packages with Covid-19 concepts

mozzy11 opened the pull request:
Ticket : https://issues.openmrs.org/browse/RA-1731

i added Covid-19 Concepts to reference metadata
cc @ibacher   @dkayiwa

dkayiwa commented on api/src/main/resources/Covid-19_Concepts-1.xml:
```diff
@@ -0,0 +1,463 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<dataset>
+<concept_datatype concept_datatype_id="1" name="Numeric" hl7_abbreviation="NM" description="Numeric value, including integer or float (e.g., creatinine, weight)" creator="1" date_created="2004-02-02 00:00:00.0" retired="false" retired_by="1" uuid="8d4a4488-c2cc-11de-8d13-0010c6dffd0f"/>
+<concept_datatype concept_datatype_id="2" name="Coded" hl7_abbreviation="CWE" description="Value determined by term dictionary lookup (i.e., term identifier)" creator="1" date_created="2004-02-02 00:00:00.0" retired="false" retired_by="1" uuid="8d4a48b6-c2cc-11de-8d13-0010c6dffd0f"/>
```
Are these new concept data types that never existed before?

mozzy11 replied:
@dkayiwa , i have removed all the existing concept data types and concept clases from this new package

mozzy11 commented:
@mozzy11  , @ibacher  , any comments on this . i want to move ahead with the release and this is a blocker

dkayiwa commented:
@mozzy11 how many new concepts did Andy supply for Covid-19?

ibacher commented:
I don't how many of these are net-new to the OpenMRS base concepts, but there are 123 concepts in the [OCL COVID-19 Starter Set](https://openconceptlab.org/orgs/CIEL/collections/COVID-19-Starter-Set/v1.0/concepts/), which is the basis for the forms we're building.

ibacher commented:
Note that a number of the COVID-related concepts are not COVID specific. We added some concepts for Reporting Institution, Reporting Country and some basic contact tracing concepts (travel, etc.) as well some general symptoms that may or may not already be in the base set.

mozzy11 commented:
@dkayiwa , i cant give the exact number really . But i simply did a search like "covid" in the MDSbuilder and i added all the concepts in that search result

dkayiwa commented:
@mozzy11 can you confirm with Andy that all concepts returned with that search text are the ones we need for the reference application?

mozzy11 commented:
Thanks @ibacher , let me also confirm from Andy

dkayiwa commented on api/src/main/resources/Reference_Application_Numeric_Concepts-25-2.x.xml:
```diff
@@ -10,4 +10,6 @@
 <concept_numeric concept_id="5092" hi_absolute="100.0" low_absolute="0.0" units="%" allow_decimal="false"/>
 <concept_numeric concept_id="5242" hi_absolute="999.0" hi_critical="120.0" hi_normal="70.0" low_absolute="0.0" units="" allow_decimal="false"/>
 <concept_numeric concept_id="5314" low_absolute="0.0" units="cm" allow_decimal="true"/>
+<concept_numeric concept_id="165836" units="" allow_decimal="true"/>
```
I thought you put the new concepts in Covid-19_Concepts-1.xml

Did you just accidentally change these other files?

mozzy11 replied:
those are some numeric concepts for Covid-19. i still wanted to keep the backward compatibility with    versions pre 2.2

dkayiwa replied:
Since you decided to separate the covid-19 in their own file, wouldn't it be consistent to also have their own concept numeric files?

mozzy11 replied:
makes sense

mozzy11 replied:
@dkayiwa , placed them in a separate file

dkayiwa commented:
@mozzy11 have you tested this by actually running the reference application with these changes and confirmed that you can search for the new concepts and see them as expected?

dkayiwa commented:
@mozzy11 the number of concepts that you have added seem to be different from the ones i get when i search for `covid` on mdsbuilder and https://openconceptlab.org/search/?q=covid

mozzy11 commented:
@dkayiwa , i added 34 concepts in total. 

- 33 concepts are the ones with a name or a Synonym "covid".
* I also added concept 155762 which is  **coronavirus infection** , it doesnt have any synonym for "covid" , so its not returned as part of the result set from that text search

mozzy11 commented:
Sure i also tested them locally

mozzy11 commented:
And also , i search from from MDSbuilder not Conceptlab. MdsBuilder returns 33 concepts from the  "covid" text search. Not Sure whether the Concept lab is updated with the very latest CIEL as MDSbuilder is

dkayiwa commented:
How did you find out about `coronavirus infection`, given that it was not returned in the covid search results?

mozzy11 commented:
i also searched "corona"

dkayiwa commented:
Of the so many results returned by corona, how did you single out only that one? Could there be more?

mozzy11 commented:
, i  looked through all of them one by one. all the other "corona" are refering to the coronary atery blood vessels.
I also went ahead to cross check with some packages for PIH ,to ensure i dont leave out any

dkayiwa commented:
How about this? 113020,

mozzy11 commented:
let me add them.

mozzy11 commented:
@dkayiwa   , the first commit i did , i had manually edited the file to remove some concepts , but this time i simply added the file as built by the metadata sharing module.   the concepts seem to be more than what appears to be in the serach , reason being ,some Covid-19 concepts are question , linked to answers which are not covid-19 related concepts, so the metadata sharing module includes all the answer Concepts

mozzy11 commented:
i have included the above two Concepts you mentioned

ibacher commented on api/src/main/resources/Covid-19_Concepts-1.xml:
```diff
@@ -0,0 +1,521 @@
+<concept_reference_source concept_source_id="21" name="CIEL" description="Columbia International eHealth Laboratory concept ID" creator="1" date_created="2013-07-03 15:55:03.0" retired="false" retired_by="1" uuid="21ADDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD" changed_by="1"/>
+<concept_reference_source concept_source_id="24" name="IMO ProblemIT" description="Intelligent Medical Objects, Inc. problem/disease/finding datasource" creator="1" date_created="2013-10-01 13:34:15.0" retired="false" retired_by="1" uuid="24ADDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD" changed_by="1"/>
+<concept_reference_source concept_source_id="25" name="IMO ProcedureIT" description="Intelligent Medical Objects, Inc. procedure/test datasource" creator="1" date_created="2013-10-01 13:34:40.0" retired="false" retired_by="1" uuid="25ADDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD" changed_by="1"/>
+<concept_reference_source concept_source_id="38" name="SNOMED UK" description="SNOMED CT UK extension" creator="1" date_created="2020-03-05 19:08:04.0" retired="false" retired_by="1" uuid="38ADDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD" changed_by="1"/>
```
A good number of these reference sources are already in the default RefApp metadata, although I'm not sure how they are loaded.

mozzy11 replied:
true @ibacher , some of them are already loaded from the Reference_Application_Concepts-25.xml file

mozzy11 replied:
let me clear them

mozzy11 replied:
@ibacher , i removed the existing reference sources

mozzy11 commented:
@dkayiwa   @ibacher   , i Updated the PR with the Covid-19 starter package as advised by Andy , Ellen and @ibacher .
here is the list https://pastebin.com/4bXz1ZKH  ,of those Concepts in the package but not in reff app already

dkayiwa commented:
@mozzy11 264 new concepts for just COVID-19?

mozzy11 commented:
@dkayiwa  , thats what i was instructed to do
https://talk.openmrs.org/t/covid-19-concept-metadata-package-for-reference-application-2-10/27734

dkayiwa commented:
@mozzy11 but the list on that link is way much less that 264

mozzy11 commented:
@dkayiwa some concepts listed there are Concept-sets  , so they come with a bunch of other Concepts as set members. 
for example concept **165657**

ibacher commented:
A big driver here is that the COVID-related concepts include case-tracking related concepts, which resulted in new concepts for every country in the world.

dkayiwa commented:
Oh i see!!! 😊
